backend/src/main/java/com/bounswe2024group10/Tradeverse/service/PortfolioService.java

```diff
@@ -0,0 +1,56 @@
+package com.bounswe2024group10.Tradeverse.service;
+
+import com.bounswe2024group10.Tradeverse.dto.portfolio.*;
+import com.bounswe2024group10.Tradeverse.model.Portfolio;
+import com.bounswe2024group10.Tradeverse.repository.PortfolioRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class PortfolioService {
+
+    @Autowired
+    private PortfolioRepository portfolioRepository;
+
+    public CreatePortfolioResponse createPortfolio(CreatePortfolioRequest request) {
+        Portfolio portfolio = new Portfolio(request.getUsername(), request.getName(), request.getVisibility());
+
+        try {
+            Portfolio savedPortfolio = portfolioRepository.save(portfolio);
+            return new CreatePortfolioResponse(true, savedPortfolio.getId(), "Portfolio created successfully");
+        } catch (Exception e) {
+            return new CreatePortfolioResponse(false, null, "Failed to create portfolio");
+        }
+    }
+
+    public GetAllPortfoliosResponse getAllPortfolios(GetAllPortfoliosRequest request) {
+        List<Portfolio> portfolios = portfolioRepository.findByUsername(request.getUsername());
+
+        if (portfolios.isEmpty()) {
+            return new GetAllPortfoliosResponse(false, "No portfolios found for this user", null);
+        }
+
+        List<PortfolioDto> portfolioDtos = portfolios.stream()
+                .map(portfolio -> new PortfolioDto(portfolio.getId(), portfolio.getName(), portfolio.getVisibility(), portfolio.getCreatedAt()))
```

Comment:
What is createdAt and why are we storing it?

It is not defined at PortfolioDto class.

Reply:
CreatedAt is removed.